src/Infrastructure/Riganti.Utils.Infrastructure.EntityFramework/EntityFrameworkUnitOfWork.cs

```diff
@@ -133,14 +156,29 @@ protected override void DisposeCore()
             if (HasOwnContext())
             {
                 Context.Dispose();
+
+                if (CommitRequested)
+                {
+                    throw new InvalidOperationException("Some of the unit of works requested commit! Ensure that commit is called on root unit of work as well.");
```

Comment:
I think we should create a concrete exception, something like `ChildCommitPendingException` and not use `InvalidOperationException`. This is very custom exception and the other reason is to check the concrete exception in the unit test `Commit_CorrectChildRequestIgnoredBehavior`, not any `InvalidOperationException`.